internal/engine/execution/global.go

```diff
@@ -156,6 +156,27 @@ func NewGlobalContext(ctx context.Context, db sql.Executor, extensionInitializer
 	return g, nil
 }
 
+// Reload is used to reload the global context based on the current state of the database.
+// It is used after state sync to ensure that the global context is up to date.
+func (g *GlobalContext) Reload(ctx context.Context, db sql.Executor) error {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+
+	schemas, err := getSchemas(ctx, db)
```

Comment:
Maybe rename this to `ReloadCache`, since it is more descriptive of what is occurring?

Also, remembering the bug Paulo ran into where a failed drop was still in memory: since we are full table scanning and reloading our cache, should we delete the cache in the first place?